CMakeLists.txt

```diff
@@ -90,6 +90,59 @@ option(NGRAPH_DEX_ONLY "Build CPU DEX without codegen" FALSE)
 option(NGRAPH_CODE_COVERAGE_ENABLE "Enable code coverage data collection" FALSE)
 option(NGRAPH_LIB_VERSIONING_ENABLE "Enable shared library versioning" FALSE)
 option(NGRAPH_PYTHON_BUILD_ENABLE   "Enable build nGraph python package wheel" FALSE)
+option(NGRAPH_PLAIDML_ENABLE "Enable the PlaidML backend" FALSE)
+if (NGRAPH_ONNX_IMPORT_ENABLE)
+    option(NGRAPH_USE_SYSTEM_PROTOBUF "Use system provided Protobuf shared object" FALSE)
+    option(NGRAPH_ONNXIFI_ENABLE "Enable ONNX Interface for Framework Integration" TRUE)
+endif()
+if (LINUX AND NGRAPH_PYTHON_BUILD_ENABLE)
+    option(NGRAPH_MANYLINUX_ENABLE   "Enable manylinux support for nGraph" FALSE)
+endif()
+
+if (NGRAPH_PYTHON_BUILD_ENABLE)
+    if(NOT NGRAPH_INTERPRETER_ENABLE)
+        message(FATAL_ERROR "Python API needs Interpreter backend!")
+    endif()
+endif()
+
+#
+# manylinux has limited support for now
+# disable all unsupported options and use DEX only for CPU backend
+#
+if (NGRAPH_MANYLINUX_ENABLE)
+    if(NGRAPH_CPU_ENABLE)
+        if(NOT NGRAPH_DEX_ONLY)
+            message(WARNING "manylinux Python API only supports Direct Execution(DEX) for CPU backend!")
+            message(WARNING "Turning on option NGRAPH_DEX_ONLY")
+            set(NGRAPH_DEX_ONLY TRUE)
+        endif()
+    endif()
+    if(NGRAPH_GPU_ENABLE)
+        message(WARNING "manylinux Python API does not support GPU backend!")
```

Comment:
All of these should be fatal errors. If the user set bad settings them tell them, don't fix them.